app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -104,13 +106,27 @@ public void onAttach(Context context) {
     }
 
     @Override
-    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+    public void onCreateOptionsMenu(final Menu menu, MenuInflater inflater) {
         switch (settingsScreen) {
             case SEARCH_ENGINES:
                 inflater.inflate(R.menu.menu_search_engines, menu);
                 break;
             case REMOVE_ENGINES:
                 inflater.inflate(R.menu.menu_remove_search_engines, menu);
+
+                // We disable the trash icon when no engine is checked, enable it otherwise
+                getView().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        MenuItem menuDeleteItem = menu.findItem(R.id.menu_delete_items);
+                        menuDeleteItem.getIcon().mutate().setAlpha(SettingsFragment.ALPHA_GREY);
```

Comment:
You shouldn't need to modify the alpha directly: this menu item uses `icon="@drawable/ic_delete"`, which has `fillColor=@drawable/state_colors` which is:

```xml
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <item android:color="#4Cffffff" android:state_enabled="false" />
    <item android:color="#ffffffff" />
</selector>
```

That is when the state is disabled, we get the disabled color. When the state matches no other clauses (i.e. is enabled), we get the enabled color.

This line, and the related lines, can change to `setEnabled(...)`

Reply:
👍  It seems not working though